contracts/core/mailbox/src/contract.rs

```diff
@@ -71,7 +71,9 @@ pub fn query(deps: Deps, env: Env, msg: QueryMsg) -> Result<QueryResponse, Contr
     match msg {
         QueryMsg::Ownable(msg) => Ok(hpl_ownable::handle_query(deps, env, msg)?),
         QueryMsg::Hook(msg) => match msg {
-            MailboxHookQueryMsg::QuoteDispatch(msg) => to_binary(quote_dispatch(deps, msg)),
+            MailboxHookQueryMsg::QuoteDispatch { sender, msg } => {
+                to_binary(quote_dispatch(deps, sender, msg))
```

Comment:
thanks for your information, i'll update it after reviewing v3 solidity contract

Reply:
so the reason why we decided to add `sender` to QueryMsg is `msg.sender` is needed to build a dispatch message according to [here](https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/4fe4000d69e810759023456ebdae269c5d1262c3/solidity/contracts/Mailbox.sol#L417)

Reply:
Of course yes. `msg` has the type `DispatchMsg` which is grouped representation of argument dispatch / quote_dispatch in solidity - it has the optional metadata field for custom use cases.
+ It's impossible to construct the mailbox message in case of quoting dispatch due to there's no way to track `msg.sender` of QueryMsg.

Reply:
this makes sense
let's make sure we are consistent everywhere we use info.sender in execute to pass through a sender in the corresponding query